packages/@sfx/sayt-driver-plugin/test/unit/browser/sayt-driver-plugin.test.ts

```diff
@@ -0,0 +1,193 @@
+            { value: 'b' },
+            { value: 'c' },
+          ],
+        },
+      };
+
+      const cbReturn = Driver.autocompleteCallback(undefined, response);
+
+      expect(cbReturn).to.deep.equal(['a', 'b', 'c']);
+    });
+
+    it('should only return entries that have a searchterm value', () => {
+      const response = {
+        result: {
+          searchTerms: [
+            { value: 'a' },
+            {},
+            { value: 'c' },
+          ],
+        },
+      };
+
+      const cbReturn = Driver.autocompleteCallback(undefined, response);
+
+      expect(cbReturn).to.deep.equal(['a', 'c']);
+    });
+
+    it('should return an empty array if there are no searchTerms', () => {
+      const response = {
+        result: {
+          searchTerms: [],
+        },
+      };
+
+      const cbReturn = Driver.autocompleteCallback(undefined, response);
+
+      expect(cbReturn).to.deep.equal([]);
+    });
+  });
+
+  describe('.sendSaytAPIRequest()', () => {
+    const saytDataPayload = {
+      query: 'shirt',
+      collection: 'backup',
+    };
+    let autocompleteCallback;
+
+    beforeEach(() => {
+      autocompleteCallback = stub(Driver, 'autocompleteCallback');
+    });
+
+    it('should make a search call through the sayt client', () => {
+      const autocomplete = spy(Driver.core.sayt, 'autocomplete');
+
+      Driver.sendSaytAPIRequest(saytDataPayload);
+
+      expect(autocomplete).to.be.calledWith(
+        saytDataPayload.query,
+        { collection: 'backup' },
+        autocompleteCallback,
+      );
+    });
+
+    it('should return the result of the Sayt API callback', () => {
+      autocompleteCallback.returns(['a', 'b']);
+      const callbackReturn = autocompleteCallback();
+
+      const returnValue = Driver.sendSaytAPIRequest(saytDataPayload);
+
+      expect(returnValue).to.eventually.equal(callbackReturn);
+    });
+  });
+
+  describe('.fetchSaytData()', () => {
+    const query = {
+      query: 'shirt',
+    };
+    let dispatchEvent;
+    let sendSaytAPIRequest;
+
+    beforeEach(() => {
+      sendSaytAPIRequest = stub(Driver, 'sendSaytAPIRequest').callThrough();
+      dispatchEvent = spy(Driver.core['dom_events'], 'dispatchEvent');
+    });
+
+    it('should get a response from Sayt client request method', () => {
+      Driver.fetchSaytData(query);
+
+      expect(sendSaytAPIRequest).to.be.calledWith(query);
+    });
+
+    it('should dispatch the response through the events plugin', () => {
+      const response = { a: 'b' };
+      sendSaytAPIRequest.returns(response);
+
+      Driver.fetchSaytData(query);
+
+      expect(Promise.resolve(dispatchEvent))
+        .to.be.eventually.calledOnceWith(Driver.saytResponseEvent, response);
+    });
+
+    it('should send an error in an event if the API request fails', () => {
+      sendSaytAPIRequest.rejects('test error');
+
+      Driver.fetchSaytData(query);
+
+      expect(Promise.resolve(dispatchEvent))
```

Comment:
Assertions using chai-as-promised should be returned so that mocha can wait on it.